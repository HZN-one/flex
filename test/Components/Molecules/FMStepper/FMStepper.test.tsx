/**
 * @jest-environment jsdom
 */
import * as React from "react";
import "@testing-library/jest-dom";
import { FMStepper } from "@Molecules";
import { render } from "@testing-library/react";

describe("FMStepper", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <FMStepper
        testID="stepper-example"
        stepperColumn={8}
        data={[
          {
            title: "Title 1",
            children: "Description 1",
            buttonLabel: "Next",
            onSubmitBack: () => {
              console.log("onSubmitBack");
            },
            onSubmitNext: () => {
              console.log("onSubmitNext");
            },
          },
          {
            title: "Title 2",
            children: "Description 2",
            buttonLabel: "Next",
            onSubmitBack: () => {
              console.log("onSubmitBack");
            },
            onSubmitNext: () => {
              console.log("onSubmitNext");
            },
          },
          {
            title: "Title 3",
            children: "Description 3",
            buttonLabel: "Next",
            onSubmitBack: () => {
              console.log("onSubmitBack");
            },
            onSubmitNext: () => {
              console.log("onSubmitNext");
            },
          },
          {
            title: "Title 4",
            children: "Description 4",
            buttonLabel: "Submit",
            onSubmitBack: () => {
              console.log("onSubmitBack");
            },
            onSubmitNext: () => {
              console.log("onSubmitNext");
            },
          },
        ]}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
