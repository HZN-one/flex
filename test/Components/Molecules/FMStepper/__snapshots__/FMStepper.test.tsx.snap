// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FMStepper should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
    data-testid="stepper-example"
  >
    <div
      class="MuiStepper-root MuiStepper-horizontal css-m5vj9m-MuiStepper-root"
    >
      <div
        class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal css-ascpo7-MuiStepLabel-root"
        >
          <span
            class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                text-anchor="middle"
                x="12"
                y="16"
              >
                1
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
          >
            <span
              class="MuiStepLabel-label Mui-active css-qivjh0-MuiStepLabel-label"
            >
              Title 1
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
        >
          <span
            class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                text-anchor="middle"
                x="12"
                y="16"
              >
                2
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
          >
            <span
              class="MuiStepLabel-label Mui-disabled css-qivjh0-MuiStepLabel-label"
            >
              Title 2
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
        >
          <span
            class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                text-anchor="middle"
                x="12"
                y="16"
              >
                3
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
          >
            <span
              class="MuiStepLabel-label Mui-disabled css-qivjh0-MuiStepLabel-label"
            >
              Title 3
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
      >
        <span
          class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
        >
          <span
            class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                text-anchor="middle"
                x="12"
                y="16"
              >
                4
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
          >
            <span
              class="MuiStepLabel-label Mui-disabled css-qivjh0-MuiStepLabel-label"
            >
              Title 4
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root css-1l6icon"
    >
      Description 1
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
          data-testid="button-stepper-example"
          tabindex="0"
          type="button"
        >
          Next
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
